src/utils.jl

```diff
@@ -154,7 +154,12 @@ specnlm2spec1p(spec_nlm)
 From a list of AA specifications to all A specifications needed
 """
 function specnlm2spec1p(spec_nlm)
-    spec1p = union(spec_nlm...)
+    # spec1p = union(spec_nlm...) # oringinal implementation which leads to StackOverflowError when input vector is too long
+    spec1p = []
```

Comment:
could you initilize this with `spec1p = spec_nlm[1]`, then you get type stability